Kudu.Core/Helpers/PostDeploymentHelper.cs

```diff
@@ -450,6 +467,63 @@ private static void WriteAutoSwapOngoing()
             }
         }
 
+        /// <summary>
+        /// This works around the hostname dns resolution issue for recently created site.
+        /// If dns failed, we will use the home hosted service as alternative IP address.
+        /// </summary>
+        /// <returns></returns>
+        private static async Task<IPAddress> GetAlternativeIPAddress(string host)
+        {
+            try
+            {
+                // if resolved successfully, return null to not use alternative ipAddress
+                await Dns.GetHostEntryAsync(host);
+                return null;
+            }
+            catch (Exception ex)
+            {
+                Trace(TraceEventType.Verbose, "Unable to dns resolve {0}.  {1}", host, ex);
+            }
+
+            return await GetHomeStampAddress(host);
+        }
+
+        private static async Task<IPAddress> GetHomeStampAddress(string host)
+        {
+            var homeStamp = HomeStamp;
+            if (string.IsNullOrEmpty(homeStamp))
+            {
+                return null;
+            }
+
+            var homeStampHostName = string.Format("{0}.cloudapp.net", homeStamp);
+            if (host.EndsWith(".scm.azurewebsites.us", StringComparison.OrdinalIgnoreCase))
+            {
+                homeStampHostName = string.Format("{0}.usgovcloudapp.net", homeStamp);
+            }
+            else if (host.EndsWith(".scm.chinacloudsites.cn", StringComparison.OrdinalIgnoreCase))
+            {
+                homeStampHostName = string.Format("{0}.chinacloudapp.cn", homeStamp);
+            }
+            else if (host.EndsWith(".scm.azurewebsites.de", StringComparison.OrdinalIgnoreCase))
+            {
+                homeStampHostName = string.Format("{0}.azurecloudapp.de", homeStamp);
+            }
+
+            try
+            {
+                Trace(TraceEventType.Verbose, "Try to dns resolve {0}.", homeStampHostName);
+                var entry = await Dns.GetHostEntryAsync(homeStampHostName);
+                return entry.AddressList.First();
+            }
+            catch (Exception ex)
+            {
+                Trace(TraceEventType.Verbose, "Unable to dns resolve {0}.  {1}", homeStampHostName, ex);
```

Comment:
This code path is a bit funky, as we'll end up trying to use the site host name, which we already know doesn't resolve. But I guess it's no worse than today...

Reply:
Yes, the intention is, if the workaround does not help, it will be as if there is no workaround at all (falling back to the old behavior).